File: backend/app/controllers/reset_controller.py
from flask import Blueprint, request, jsonify
from services.reset_service import reset_user_password

reset_bp = Blueprint("reset", __name__)

@reset_bp.route("/reset-password/<password_verify_code>", methods=["POST"])
def reset_password(password_verify_code):
    data = request.json
    new_password = data.get("password")
    if not new_password:
        return jsonify({"message": "請提供新密碼"}), 400

    result, error = reset_user_password(password_verify_code, new_password)
    if error:
        return jsonify({"message": error}), 400

    return jsonify({"message": result}), 200

File: backend/nest-app/src/controllers/reset.controller.ts
import {
    Controller,
    Post,
    Param,
    Body,
    BadRequestException,
  } from '@nestjs/common';
  import { ResetService } from '../services/reset.service';

  @Controller('reset-password')
  export class ResetController {
    constructor(private readonly resetService: ResetService) {}

    @Post(':passwordVerifyCode')
    async resetPassword(
      @Param('passwordVerifyCode') code: string,
      @Body('password') newPassword: string,
    ) {
      if (!newPassword) {
        throw new BadRequestException('請提供新密碼');
      }

      const result = await this.resetService.resetUserPassword(code, newPassword);
      if (result.error) {
        throw new BadRequestException(result.error);
      }

      return { message: result.message };
    }
  }
